generate_engagement_analysis_files.py

```diff
@@ -0,0 +1,103 @@
+from collections import OrderedDict
+import argparse
+import json
+import csv
+
+from core_data_modules.logging import Logger
+
+from src.lib import PipelineConfiguration
+
+Logger.set_project_name("IMAQAL")
+log = Logger(__name__)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Computes engagement data and stores it in multiple CSV sheets for"
+                                                 "analysis and visualization")
+
+    parser.add_argument("user", help="User launching this program")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file",
+                        help="Path to the pipeline configuration json file"),
+    parser.add_argument("demog_map_json_input_dir", metavar="demog-map-json-input-dir",
+                        help="Path to a directory to read per episode demog map .json files. "
+                             "The files should be in the format <rqa_raw_field>_demog_map.json")
+    parser.add_argument("engagement_csv_output_dir", metavar="engagement-csv-output-dir",
+                        help="Directory to write engagement CSV files to")
+
+
+    args = parser.parse_args()
+
+    user = args.user
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    demog_map_json_input_dir = args.demog_map_json_input_dir
+    engagement_csv_output_dir = args.engagement_csv_output_dir
+
+    # Load the pipeline configuration file
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+
+    # Infer which RQA coding plans to use from the pipeline name.
+    if pipeline_configuration.pipeline_name == "q4_pipeline":
+        log.info("Running Q4 pipeline")
```

Comment:
These don't actually run the pipeline though do they? They extract data from a previous run of the pipeline to produce a CSV file

Reply:
That's right have updated this to `log.info("Extracting Q4 pipeline data")`